ports/raspberrypi/common-hal/nvm/ByteArray.c

```diff
@@ -42,6 +43,9 @@ uint32_t common_hal_nvm_bytearray_get_length(const nvm_bytearray_obj_t *self) {
 }
 
 static void write_page(uint32_t page_addr, uint32_t offset, uint32_t len, uint8_t *bytes) {
+    // disable interrupts to prevent core hang on rp2040
+    common_hal_mcu_disable_interrupts();
```

Comment:
Can move this down to turn off interrupts only on the actual flash operation. The `memcpy()` doesn't need to be protected.
```suggestion
```